lib/helpers.js

```diff
@@ -137,37 +137,40 @@ function isMasterPreviewAccessible(courseId, userId, next) {
 
 function hasCoursePermission (action, userId, tenantId, contentItem, next) {
   // Check that the contentItem has something resembling a courseId
-  if (contentItem && typeof contentItem._id === 'undefined' && typeof contentItem._courseId === 'undefined') {
+  if (contentItem && !contentItem._id && !contentItem._courseId) {
     // Course permission cannot be verified
     return next(null, false);
   }
-
-  var courseId = contentItem && contentItem._courseId
-      ? contentItem._courseId.toString()
-      : contentItem._id.toString();
-
   database.getDatabase(function(err, db) {
     if (err) {
       logger.log('error', err);
       return err;
     }
+    const courseId = contentItem && contentItem._courseId ? contentItem._courseId.toString() : contentItem._id.toString();
 
     db.retrieve('course', { _id: courseId }, { jsonOnly: true }, function (err, results) {
       if (err) {
         logger.log('error', err);
         return next(err);
       }
       if (!results || results.length !== 1) {
-        return next(new Error('Course ' + courseId + ' not found'));
+        return next(new Error(`Course ${courseId} not found`));
       }
+      const course = results[0];
 
-      var course = results[0];
-
-      if ((course._isShared || course.createdBy.toString() == userId) && course._tenantId.toString() == tenantId) {
-        // Shared courses on the same tenant are open to users on the same tenant
+      if(course._tenantId.toString() !== tenantId) { // no sharing across tenants
+        return next(null, false);
+      }
+      if(course.createdBy.toString() === userId) { // user's own course
```

Comment:
extra line above this

Reply:
Sorry, is this asking to remove the line break?

Reply:
I don't know :upside_down_face: happy to resolve